DMVM-204(build): xcconfig 설정

bokoo14 opened the pull request:
## 🎫 지라 티켓 이슈
https://dogbusiness.atlassian.net/browse/DMVM-204

<br>

<br>

## 🛠️ 작업 내용
<!-- 작업 내용 (스크린샷도 같이 있으면 좋아요) -->
dev환경과 release환경을 분리하기 위해 xcconfig 파일과 info.plist파일을 사용합니다.

## 기존
기존의 작업 환경에서는 API를 gitignore에 설정해두어 함께 작업하기 위해 수작업으로 파일을 해당하는 디렉토리에 저장해야 하여 불편함이 있었습니다.
## 작업 후
xcconfig파일을 활용하여 HOST뿐만 아니라, AppIcon, APP_BUNDLE_ID, WEB_BASE_URL 등을 모두 다룰 수 있어 dev환경과 release환경을 쉽게 분리할 수 있게 되었습니다.
추후 BE에서 dev, release환경이 분리되면 xcconfig에서 HOST주소만 수정하면 되어, 작업이 용이해졌습니다. 
/Mongle/Projects/Core/Sources/API -> 제거하시면 됩니다


## 기본 지식
xcconfig에 대한 설명 및 적용 방법은 해당 컨플 문서에 정리해두었으니 참고 바랍니다. 
https://dogbusiness.atlassian.net/wiki/spaces/TEAM/pages/38502405/xcconfig+dev+release

## 적용방법
<img width="654" alt="스크린샷 2024-09-18 오후 9 20 42" src="https://github.com/user-attachments/assets/64b9db13-8923-4b10-b152-1eb47ef2dbbc">

xcconfig 파일을 해당 경로에 저장 후, tuist에서 작업합니다.

<img width="618" alt="스크린샷 2024-09-18 오후 9 26 43" src="https://github.com/user-attachments/assets/8ed116ca-4d3a-4d45-ac31-e1df579fbd1c">

settings의 configurations에서 debug와 release 환경에서 xcconfig 파일을 지정해줍니다. 

앱 모듈에서 해당 settings를 적용해줍니다. 
또한 xcconfig파일에 적용된 변수들을 사용하기 위해서는 info.plist파일에서도 아래와 같이 적용해주어야 합니다.
<img width="572" alt="스크린샷 2024-09-18 오후 9 31 54" src="https://github.com/user-attachments/assets/cd50d5b4-9dcc-48b3-bcf2-9b1adb6c5d26">
앱모듈에 위와 같이 적용된 것을 확인할 수 있습니다.
<img width="407" alt="스크린샷 2024-09-18 오후 9 30 00" src="https://github.com/user-attachments/assets/1ddb3b4e-a9be-4036-ab6b-266f8043424f">

## 사용방법
Core 모듈에 있는 Config 파일 > Config와 ConfigImp를 이용하여 작업할 수 있습니다.
기존의 Network관련 코드에서는 API를 직접 작성하였지만, configImp를 사용하여 baseURL을 가져올 수 있게 됩니다. 
<img width="1085" alt="스크린샷 2024-09-18 오후 9 34 07" src="https://github.com/user-attachments/assets/eff97d78-52df-4888-bf3a-85acdd2aadfc">

## 참고 사항
추후 Network 관련하여 generic하게 사용할 수 있도록 코드 추가 작업이 필요합니다. 

<br>

<br>

## 📱 작업 화면
<!-- img src "이부분에 gif파일 넣어주시면 됩니다" -->
|화면 이름|스크린샷|
|:--:|:--:|
|ViewName|<img src = "" width ="250">|

<br>

## 🗒️ Note (optional)
<!-- 추가 필요한 사항이나 하고픈 말
     Reviewer 한테 요청하고 싶은 것들
     코드리뷰 요청하고 싶은 것들.. 등등 -->

bokoo14 committed:
DMVM-204(build): xcconfig 구성

bokoo14 committed:
DMVM-204(chore): gitignore 수정

bokoo14 committed:
DMVM-204(refactor): dev, release 분리 및 코드 정리

bokoo14 committed:
DMVM-204(fix): infoPlist오류 수정

bulmang reviewed (approved):
고생하셨습니다. setting/xcconfig branch에서는 dev, release config 파일 설정이 잘 작동되나요?

bokoo14 commented:
> 고생하셨습니다. setting/xcconfig branch에서는 dev, release config 파일 설정이 잘 작동되나요?

흠 .. 🧐
제가 작업한 바로는 tuist로 config 설정 완료했다고 생각하고 있는데, 뭔가 찝찝한 점이 있어서 확인하고 다시 멘션드릴게용

bulmang commented:
> > 고생하셨습니다. setting/xcconfig branch에서는 dev, release config 파일 설정이 잘 작동되나요?
> 
> 흠 .. 🧐 제가 작업한 바로는 tuist로 config 설정 완료했다고 생각하고 있는데, 뭔가 찝찝한 점이 있어서 확인하고 다시 멘션드릴게용

알겠습니다~~ 추석에 파이팅 넘치시는군요! 👍

bokoo14 committed:
DMVM-204(chore): dev용 Icon 추가

bokoo14 committed:
DMVM-204(chore): xcconfig 수정

bokoo14 committed:
DMVM-204(feat): xcconfig 사용 로직 추가

bokoo14 committed:
DMVM-204(refactor): networkmanager xcconfig 사용

bokoo14 committed:
DMVM-204(chore): xcconfig 파일명 수정

bokoo14 committed:
DMVM-204(feat): dev, release 환경 분리

-  "PRODUCT_BUNDLE_IDENTIFIER"
- "PROVISIONING_PROFILE_SPECIFIER"
- "CODE_SIGN_IDENTITY"

dev, release에 따라 환경을 분리합니다

bokoo14 committed:
DMVM-204(refactor): API코드 제거

bokoo14 committed:
DMVM-204(feat): dev, release bundle id setting 분리

bulmang commented on Mongle/Tuist/ProjectDescriptionHelpers/Project+Templates.swift:
```diff
@@ -14,29 +14,48 @@ public extension Project {
-        let bundleSuffix = "dev"
         let deploymentTarget = Environment.deploymentTarget
         var projectTargets: [Target] = []
+        
+        // MARK: TARGETS > Singing & Capabilities
+        let devSigningSettings: [String: SettingValue] = [
+            "PRODUCT_BUNDLE_IDENTIFIER": "com.mongle.dev",
+            "PROVISIONING_PROFILE_SPECIFIER": "Mongle.dev",
+            "CODE_SIGN_IDENTITY": "\(APIKey.codeSignIdentity)"
+        ]
+        let releaseSigningSettings: [String: SettingValue] = [
+            "PRODUCT_BUNDLE_IDENTIFIER": "com.mongle.release",
+            "PROVISIONING_PROFILE_SPECIFIER": "Mongle.release",
+            "CODE_SIGN_IDENTITY": "\(APIKey.codeSignIdentity)"
+        ]
```
```suggestion
        
        enum ConfigurationType: String {
            case dev = "dev"
            case release = "release"
        }

        func signingSettings(for config: ConfigurationType) -> [String: SettingValue] {
            return [
                "PRODUCT_BUNDLE_IDENTIFIER": "com.mongle.\(config.rawValue)",
                "PROVISIONING_PROFILE_SPECIFIER": "Mongle.\(config.rawValue)",
                "CODE_SIGN_IDENTITY": "\(APIKey.codeSignIdentity)"
            ]
        }
```
enum Type을 만들어서 `.dev`와 `.release`를 받아 SettingValue를 반환 해주는 `signingSettings`함수를 만들면 어떨까요?

bokoo14 replied:
오! 좋은거같습니다 ㅎㅎ 👍🏻

bulmang commented on Mongle/Tuist/ProjectDescriptionHelpers/Project+Templates.swift:
```diff
@@ -14,29 +14,48 @@ public extension Project {
                 "CODE_SIGN_IDENTITY": "\(APIKey.codeSignIdentity)"
             ],
             configurations: [
+                .debug(
+                    name: .configuration("Debug"),
+                    settings: devSigningSettings,
+                    xcconfig: .relativeToRoot("Config/Debug.xcconfig")
+                ),
+                .release(
+                    name: .configuration("Release"),
+                    settings: releaseSigningSettings,
+                    xcconfig: .relativeToRoot("Config/Release.xcconfig")
+                )
```
위의 `signingSettings`함수를 사용한다면 아래처럼 함수를 호출하여 사용할 수 있습니다.
```suggestion
                .debug(
                    name: .configuration("Debug"),
                    settings: signingSettings(for: .dev),
                    xcconfig: .relativeToRoot("Config/Debug.xcconfig")
                ),
                .release(
                    name: .configuration("Release"),
                    settings: signingSettings(for: .release),
                    xcconfig: .relativeToRoot("Config/Release.xcconfig")
                )
```

bokoo14 committed:
DMVM-204(chore): signing 설정 분리 및 코드 리팩토링

Co-authored-by: bulmang <114594496+bulmang@users.noreply.github.com>

DMVM-204(chore): signing 설정 분리 및 코드 리팩토링

Co-authored-by: bulmang <114594496+bulmang@users.noreply.github.com>
